lib/Sema/CSDiagnostics.cpp

```diff
@@ -3528,6 +3527,66 @@ ContextualFailure::getDiagnosticFor(ContextualTypePurpose context,
   return None;
 }
 
+bool NonClassTypeToAnyObjectConversionFailure::diagnoseAsError() {
+  if (RawLocator->isForContextualType()) {
+    ContextualFailure failure(getSolution(), RawFromType, RawToType,
+                              RawLocator);
+    return failure.diagnoseAsError();
+  }
+
+  if (RawLocator->isLastElement<LocatorPathElt::ApplyArgToParam>()) {
+    ArgumentMismatchFailure failure(getSolution(), RawFromType, RawToType,
+                                    RawLocator);
+    return failure.diagnoseAsError();
+  }
+
+  Optional<Diag<Type, Type>> diagnosis;
+  bool forProtocol = RawToType->isConstraintType();
+  auto originalAnchor = getLocator()->getAnchor();
```

Comment:
I'm assuming that you was a "raw" anchor, if so then `FailureDiagnostic` already has a facility for that - `getRawAnchor()`.